pkg/volume/csi/csi_plugin.go

```diff
@@ -892,35 +891,28 @@ func highestSupportedVersion(versions []string) (*utilversion.Version, error) {
 		return nil, errors.New(log("CSI driver reporting empty array for supported versions"))
 	}
 
-	// Sort by lowest to highest version
-	sort.Slice(versions, func(i, j int) bool {
-		parsedVersionI, err := utilversion.ParseGeneric(versions[i])
-		if err != nil {
-			// Push bad values to the bottom
-			return true
-		}
-
-		parsedVersionJ, err := utilversion.ParseGeneric(versions[j])
-		if err != nil {
-			// Push bad values to the bottom
-			return false
-		}
-
-		return parsedVersionI.LessThan(parsedVersionJ)
-	})
-
+	var highestSupportedVersion *utilversion.Version
+	var theErr error
 	for i := len(versions) - 1; i >= 0; i-- {
-		highestSupportedVersion, err := utilversion.ParseGeneric(versions[i])
+		currentHighestVer, err := utilversion.ParseGeneric(versions[i])
 		if err != nil {
-			return nil, err
+			theErr = err
+			continue
 		}
-
-		if highestSupportedVersion.Major() <= 1 {
-			return highestSupportedVersion, nil
+		if currentHighestVer.Major() > 1 {
```

Comment:
@saad-ali do you know why we are skipping major version > 1? Do we want to make it a const?

Reply:
If we define a constant, would it be defined in semver.go ?

@saad-ali 
What do you think ?

Reply:
I believe the logic here was that CSI currently only has version 0.x and 1.x (see https://github.com/container-storage-interface/spec/releases). And therefore any driver claiming version 2.x+ is ignored as an unsupported versions. This is ok because future 1.x versions of CSI are supposed to be backwards compatible so this version of Kubernetes will work with any 1.x driver (or 0.x), but it may not work with 2.x drivers (because 2.x does not have to be backwards compatible with 1.x).

Making this a const, SGTM. Also, maybe add a comment capturing this, because I'm sure others reading the code will have the same question in the future.